tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleCore.java

```diff
@@ -212,6 +282,32 @@ public boolean visit( IResourceDelta delta ) throws CoreException
                                 configureIfLiferayProject( project, GradleCore.this );
                             }
 
+                            // for only delete bundles dir
+                            if( delta.getKind() == IResourceDelta.REMOVED )
+                            {
+                                IResource deletedRes = delta.getResource();
+
+                                IProject project = deletedRes.getProject();
+
+                                IPath bundlesPath = project.getFullPath().append( "bundles" );
```

Comment:
The name of the directory is configurable.  So if a directory is being deleted ( it should only check for directories not files) and then if that directory is a direct child of a "workspace" directory, then we should check gradle.properties for the directory of the bundles and see if it can be matched to this resource being deleted.  Then we can delete the servers.  Also we need to protect all of these "listeners" in separate try/catch so that if one fails it doesn't break the others.